File: app/src/main/java/org/cuberite/android/preferences/MaterialListPreference.java
package org.cuberite.android.preferences;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.preference.ListPreference;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;

public class MaterialListPreference extends ListPreference {
    public MaterialListPreference(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onClick() {
        final Context context = getContext();
        new MaterialAlertDialogBuilder(context)
                .setTitle(getTitle())
                .setCancelable(true)
                .setSingleChoiceItems(
                        getEntries(),
                        findIndexOfValue(getValue()),
                        (dialog, index) -> {
                            if (callChangeListener(getEntryValues()[index].toString())) {
                                setValueIndex(index);
                            }
                            dialog.dismiss();
                        })
                .setNegativeButton(
                        getNegativeButtonText(),
                        (dialog, button) -> dialog.dismiss())
                .show();
    }
}

File: app/src/main/kotlin/org/cuberite/android/preferences/MaterialListPreference.kt
package org.cuberite.android.preferences

import android.content.Context
import android.content.DialogInterface
import android.util.AttributeSet
import androidx.preference.ListPreference
import com.google.android.material.dialog.MaterialAlertDialogBuilder

class MaterialListPreference(context: Context, attrs: AttributeSet?) : ListPreference(context, attrs) {
    override fun onClick() {
        MaterialAlertDialogBuilder(context)
                .setTitle(title)
                .setCancelable(true)
                .setSingleChoiceItems(
                        entries,
                        findIndexOfValue(value)
                ) { dialog: DialogInterface, index: Int ->
                    if (callChangeListener(entryValues[index].toString())) {
                        setValueIndex(index)
                    }
                    dialog.dismiss()
                }
                .setNegativeButton(
                        negativeButtonText
                ) { dialog: DialogInterface, _: Int -> dialog.dismiss() }
                .show()
    }
}
